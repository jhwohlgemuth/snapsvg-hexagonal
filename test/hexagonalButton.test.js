'use strict';

const {mockHexagonalButton} = require('./mocks');
const HexagonalButton = require('../lib/hexagonalButton');

Object.assign(HexagonalButton.prototype, mockHexagonalButton);

describe('Hexagonal Button', function() {
    let button;
    let sideLength = 10;
    let width = 100;
    let level = 0;
    let title = 'Test Button';
    let subtitle = 'Ready for testing';
    beforeEach(() => {
        button = new HexagonalButton({sideLength, width, level, title, subtitle});
    });
    it('can be created with correct points', () => {
        expect(button).toMatchSnapshot();
        expect(button.addClass.mock.calls).toMatchSnapshot();
        expect(button.attr.mock.calls).toMatchSnapshot();
    });
    it('can enable and disable hover animation', function() {
        expect(button.hoverAnimationDisabled).toBeFalsy();
        button.disableHoverAnimation();
        expect(button.hoverAnimationDisabled).not.toBeFalsy();
        button.enableHoverAnimation();
        expect(button.hoverAnimationDisabled).toBeFalsy();
    });
});
