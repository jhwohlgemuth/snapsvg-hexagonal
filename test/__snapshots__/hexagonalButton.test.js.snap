// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hexagonal Button can be created with correct points 1`] = `
 {
  "bar": [Circular],
  "blur": [Function],
  "disable": [Function],
  "disableHoverAnimation": [Function],
  "enable": [Function],
  "enableHoverAnimation": [Function],
  "focus": [Function],
  "height": 15,
  "hoverAnimationDisabled": false,
  "subTitle": [Circular],
  "title": [Circular],
  "width": 100,
}
`;

exports[`Hexagonal Button can be created with correct points 2`] = `
Array [
  Array [
    "menu-item-title",
  ],
  Array [
    "menu-item-subtitle",
  ],
  Array [
    "hexagonalButton menu-item",
  ],
]
`;

exports[`Hexagonal Button can be created with correct points 3`] = `
Array [
  Array [
    Object {
      "fill": "#FFF",
      "fontSize": 5,
      "textAnchor": "middle",
      "x": "+=55",
      "y": "+=8.75",
    },
  ],
  Array [
    Object {
      "fill": "#FFF",
      "fontSize": 3.5,
      "opacity": 0,
      "textAnchor": "middle",
      "x": 8.660254037844387,
      "y": "+=11.25",
    },
  ],
  Array [
    Object {
      "fill": "#222",
      "stroke": "#222",
      "strokeWidth": "1px",
    },
  ],
  Array [
    Object {
      "style": "-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;outline:none;",
    },
  ],
]
`;
